src/pages/EntityDetails/Model/ActionLogs/ActionLogs.tsx

```diff
@@ -0,0 +1,212 @@
+import { useEffect, useMemo, useState } from "react";
+import { DefaultRootState, useSelector, useStore } from "react-redux";
+import { useParams, Link } from "react-router-dom";
+
+import ModularTable from "@canonical/react-components/dist/components/ModularTable/ModularTable";
+
+import { getModelUUID, getModelStatus } from "app/selectors";
+import { queryOperationsList } from "juju/index";
+import { generateIconImg, generateStatusElement } from "app/utils/utils";
+
+import type { EntityDetailsRoute } from "components/Routes/Routes";
+
+type ApplicationList = { [key: string]: any };
+
+type Operations = Operation[];
+
+type Operation = {
+  operation: string;
+  actions: Action[];
+};
+
+type TableRows = TableRow[];
+
+type TableRow = {
+  application: string;
+  id: string;
+  status: string;
+  taskId: string;
+  message: string;
+  completed: string;
+};
+
+// https://github.com/juju/js-libjuju/blob/master/api/facades/action-v6.ts#L27
+type Action = {
+  action: ActionData;
+  enqueued: string;
+  started: string;
+  completed: string;
+  status: string;
+  message: string;
+};
+
+type ActionData = {
+  tag: string;
+  receiver: string;
+  name: string;
+};
+
+function generateLinkToApp(
+  appName: string,
+  userName: string,
+  modelName: string
+) {
+  return (
+    <Link to={`/models/${userName}/${modelName}/app/${appName}`}>
+      {appName}
+    </Link>
+  );
+}
+
+export default function ActionLogs() {
+  const [operations, setOperations] = useState<Operations>([]);
+  const { userName, modelName } = useParams<EntityDetailsRoute>();
+  const appStore = useStore();
+  const getModelUUIDMemo = useMemo(() => getModelUUID(modelName), [modelName]);
+  // Selectors.js is not typescript yet and it complains about the return value
+  // of getModelUUID. TSFixMe
+  const modelUUID = useSelector(
+    getModelUUIDMemo as (state: DefaultRootState) => string
+  );
+  const modelStatusData = useSelector(
+    getModelStatus(modelUUID) as (
+      state: DefaultRootState
+    ) => { applications: ApplicationList }
+  );
+
+  const applicationList = () => Object.keys(modelStatusData.applications);
+
+  useEffect(() => {
+    async function fetchData() {
```

Comment:
This will redefine this function each time the data changes right?

Reply:
It will redefine it every time the `modelUUID` changes, which should be impossible without a page change anyways. You can see the note below with the `XXX` with an issue about it fetching the data too often - this is something we'll have to follow-up on to resolve.